Tests/ClangPlugins/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+find_package(Clang 16 CONFIG REQUIRED)
+find_package(LLVM 16 CONFIG REQUIRED)
+include(AddLLVM)
+
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+get_property(CLANG_PLUGINS_ALL_COMPILE_OPTIONS GLOBAL PROPERTY CLANG_PLUGINS_ALL_COMPILE_OPTIONS)
+list(APPEND CLANG_PLUGINS_ALL_COMPILE_OPTIONS -std=c++20 -Wno-user-defined-literals -Wno-literal-range)
+
+get_property(CLANG_PLUGINS_INCLUDE_DIRECTORIES TARGET AK PROPERTY INCLUDE_DIRECTORIES)
+list(APPEND CLANG_PLUGINS_INCLUDE_DIRECTORIES ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES})
+
+configure_lit_site_cfg(
+    ${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.py.in
+    ${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg.py
+    MAIN_CONFIG ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
+    PATHS
+        LLVM_BINARY_DIR
+        LLVM_TOOLS_DIR
+        LLVM_LIBS_DIR
+        CMAKE_LIBRARY_OUTPUT_DIRECTORY
+        CMAKE_CURRENT_SOURCE_DIR
+)
+
+add_custom_command(
+    OUTPUT venv
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/requirements.txt requirements.txt
+    COMMAND ${Python3_EXECUTABLE} -m venv venv
+    COMMAND ./venv/bin/pip install -r requirements.txt --upgrade
+)
+add_custom_target(TestClangPluginsDependencies ALL
+    DEPENDS venv AllClangPlugins
+    SOURCES requirements.txt
+)
+add_test(
+    NAME TestClangPlugins
+    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/venv/bin/lit -v .
+)
```

Comment:
@nico is there a more ergonomic way to run lit tests from CMake? This looks like it will work, but feels a bit ... awkward..

Reply:
The only other thing I could find was `add_lit_target`, but I couldn't really shove that into a test target, and even if I could I think it would be just as ugly anyways, as the `venv` initialization would still have to be an `ALL` target.

Reply:
I guess the question was meant to be more about how to get lit on the user's machine in a useful place. but out of the llvm tree, creating a venv seems all right i s'pose.